src/utils/hoc.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import _ from 'lodash/fp'
+import StripedLoader from './StripedLoader'
+
+export let withNode = (Component, { allowEmptyNode = false } = {}) => ({
+  tree,
+  node,
+  path,
+  ...props
+}) => {
+  node = node || (tree && path && tree.getNode(path))
+
+  if (!node && !allowEmptyNode)
```

Comment:
Is there any reason to support `allowEmptyNode`? it's the only reason for this HOC to have config and I don't see why this would ever be ok.

Reply:
you're right, this was an oversight - I assumed it wasn't safe to take out (yet) because of the inject stuff in QueryBuilder/FilterList, but I just checked and it looks like it's fine even with that still in 👍